src/SerialPort/SerialPort.ts

```diff
@@ -86,11 +89,31 @@ export const createSerialPort = async (
         ipcRenderer.invoke(SERIALPORT_CHANNEL.SET, path, newOptions);
     };
 
-    const error = await ipcRenderer.invoke(
-        SERIALPORT_CHANNEL.OPEN,
-        options,
-        overwriteOptions
-    );
+    const tryAndOpen = async (retryCount: number) => {
```

Comment:
This is a bit nitpicky, but could you rename the function? `tryAndOpen`, sounds like you try something and then you open. I suggest `tryOpen`, or `openWithRetries`.